BlueDBOnDisk/src/test/java/org/bluedb/disk/ReadableDbOnDiskTest.java

```diff
@@ -754,6 +765,25 @@ public void test_getAllCollectionsFromDisk() throws Exception {
         assertEquals(7, allCollections.size());
 	}
 
+	@Test
+    public void test_getAllCollectionsFromDiskWithNullKeyType() throws Exception {
+        try {
+	        db().getCollectionBuilder("null_key", null, String.class).build();
+	        fail("Building a new collection with a null key type should have thrown an exception");
+        } catch(BlueDbException e) {
+        	//This is what we expect
+        }
+		
+        ReadWriteDbOnDisk newDatabase = (ReadWriteDbOnDisk) new BlueDbOnDiskBuilder().withPath(dbPath).build();

```

Comment:
BlueDB caches references to the collections, so you have to build a new bluedb instance to get it to get it to actually initialize them from disk.